sources/scraping/__init__.py

```diff
@@ -0,0 +1,72 @@
+"""Scraping source using scrapy"""
+from typing import Any, Iterator, List, TypeVar
+from queue import Empty
+
+import dlt
+
+from dlt.common import logger
+from dlt.common.configuration.inject import with_config
+
+from .helpers import ScrapingConfig
+from .queue import BaseQueue, QueueClosedError
+
+
+T = TypeVar("T")
+
+
+@dlt.source
+def scrapy_source(queue: BaseQueue[T]) -> Iterator[Any]:
```

Comment:
I do not think we need to expose any of those. user needs only your `run` function for interactions.